### PR TITLE
Feature/1124 Added Pytorch Lightning Test support for all models

### LINKED ISSUE #1124: Add support for Pytorch Lightning Trainer.test
## 🚀 Feature
**Is your feature request related to a problem? Please describe.**
Currently, it is not possible to test an IceVision model with Pytorch Lightning (we can only train it but cannot test it). 

**Describe the solution you'd like**
We should override the `test_step` method in the IceVision Pytorch Lightning Model Adapter class so that it can be used by the Pytorch Lightning Trainer class (see https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#testing ) 

**Describe alternatives you've considered**
I've considered using the `trainer.validate` since IceVision already overrides the `validation_step` but this is not ideal since the validation step logs using [`self.log("val_loss", loss)`](https://github.com/airctic/icevision/blob/356d29ba3ddc0d51c88e6c4ed78ac9bfb4dde4cc/icevision/models/ultralytics/yolov5/lightning/model_adapter.py#L58) which means that test metrics would be grouped with validaton metrics (eg: Using wandb logging).   
This is not ideal since we lose the ability to clearly distinct validation metrics from test metrics.

**Additional context**
PL Error : 
```
MisconfigurationException: No `test_step()` method defined to run `Trainer.test`.
```

### PR DESCRIPTION
**Pull request by AlexandreBrown**
This closes #1124  
- [x] Update models PL model adapter
- [x] Unit Tests 
- [x] Update notebook documentation  
- [x] End-to-end test in notebook 
image

### CODE REVIEW

**Comment by AlexandreBrown** [comment]
I added some additional unit tests to also test the `on_test_epoch_end`, `on_validation_epoch_end` and the `validate` method from the Pytorch Lightning `Trainer.validate` to ensure we can validate as well.  

If someone can re-run the github workflows I would appreciate it (I'm a first-time contributor so I cannot trigger it).

**Comment by AlexandreBrown** [comment]
@fstroth @ai-fast-track @FraPochetti The unit tests are passing on my local machine but not on the CI (OOM Killed).  
This looks like a memory issue, this PR adds a lot of tests where a lot of models are created, we had a similar issue last week on our side and needed to run the tests on a machine with more RAM.
Is there a way to fix this ?

**Comment by ai-fast-track** [comment]
> 

Yes, we do experience that now and then. It's good to know the tests passed on your local machine.

Thank you for your contribution!

**Comment by AlexandreBrown** [comment]
@ai-fast-track It's a pleasure to contribute, we're really enjoying IceVision so far.  
Let me know what is left to get it merged, we're in great need of these changes so just let me know and I'll get it done if anything.  
Cheers!

**Review by fstroth — APPROVED** [review]
Looks good to me, really nice PR. Thank you a lot @AlexandreBrown!

**Comment by potipot** [comment]
This is really nice, very useful PR.

**Comment by AlexandreBrown** [comment]
Here is a photo showing the tests passing locally, we can also see that the RAM usage was steady (~13GB).  
Screenshot from 2022-07-26 08-18-01

**Comment by potipot** [comment]
If this PR prevents test from passing in the future we should tackle this OOM bug asap. Otherwise gh ci/cd will be misleading
